packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -151,10 +180,10 @@ export class RestrictedControllerMessenger<
    * @param action - The action type. This is a unqiue identifier for this action.
    * @param handler - The action handler. This function gets called when the `call` method is
    * invoked with the given action type.
-   * @throws Will throw when a handler has been registered for this action type already.
+   * @throws Will throw if an action handler that is not in the current namespace is being registered.
    * @template T - A type union of Action type strings that are namespaced by N.
    */
-  registerActionHandler<T extends Namespaced<N, Action['type']>>(
+  registerActionHandler<T extends Action['type']>(
```

Comment:
It looks like this type no longer requires the action to be in the current namespace. Was that intentional?

Reply:
e.g. maybe it should be

```suggestion
  registerActionHandler<T extends NamespacedName<N>>(
```

Reply:
Oh great catch! I forgot to restore this constraint. Fixed: d84ade9ffa9ca59311155b1876fdf921097815cc